### PR TITLE
Remove FluxMapStore

### PR DESCRIPTION
**Pull request by alexeyraspopov**
As discussed in #343 let's remove `FluxMapStore` that has ImmutableJS as a dependency and add new guide on how to create new types of store based on `FluxStore` and `FluxReduceStore`.

@kyldvs should I push changes to `dist` folder as well?

Next things to do:

 - [ ] Provide a guide on how to extend `FluxStore` and `ReduceStore`
 - [ ] Migration guide (or small explanation, at least) for those who will need to convert their MapStores to Reduce ones

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 ### 3.0.0
 
+* `FluxMapStore`: Removed in favor to `FluxReduceStore`
```
Comment (kyldvs) [inline CHANGELOG.md]: Let's change this to:

```
* `FluxMapStore`: Removed. It added very little value over `FluxReduceStore`.
```

File: src/stores/__tests__/FluxMapStore-test.js [hunk 2]
```diff
@@ -1,116 +0,0 @@
-/**
```
Comment (kyldvs) [inline src/stores/__tests__/FluxMapStore-test.js]: If you don't mind, I think now FluxReduceStore is basically untested. This indirectly tested it. Could you just translate this as a test file for FluxReduceStore?

Comment (alexeyraspopov) [inline src/stores/__tests__/FluxMapStore-test.js]: Sure!

**Comment by alexeyraspopov** [comment]
@kyldvs I've updated tests for FluxMapStore so they now used for FluxReduceStore. Probably we could add more tests for some edge cases but in separate PR.

**Comment by kyldvs** [comment]
Awesome, thanks for the help on this stuff @alexeyraspopov!

**Comment by allizad** [comment]
Were there any guides created on "how to extend FluxStore and ReduceStore"?

**Comment by Daniel15** [comment]
Instead of removing FluxMapStore, perhaps we should have just moved it into a separate npm package (like `flux-immutable` or something)
